frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -2173,6 +2215,22 @@ def encrypt(self, resulting_type: ConversionType, x: Conversion) -> Conversion:
     def equal(self, resulting_type: ConversionType, x: Conversion, y: Conversion) -> Conversion:
         return self.comparison(resulting_type, x, y, accept={Comparison.EQUAL})
 
+    def shift_left(self, x: Conversion, rank: int) -> Conversion:
+        assert rank >= 0
+        assert rank < x.bit_width
+        shifter = 2**rank
+        shifter = self.constant(self.i(x.bit_width + 1), shifter)
+        return self.mul(x.type, x, shifter)
+
+    def reduce_precision(self, x: Conversion, bit_width: int) -> Conversion:
+        assert bit_width > 0
+        assert bit_width <= x.bit_width
```

Comment:
Maybe an assertion on the original bit width? If it makes sense I mean.

Reply:
Let's say we have a 1b value encoded on 10bits. you want to ensure the 1b are never lost ? or to prevent any overflow ?
e.g. :
`assert bit_width >= x.original_bit_width`

Reply:
Yes, that assertion could be nice to prevent overflows I think.

Reply:
But shouldn't it be more like:
```
assert rank <= x.bit_width - x.original_bit_width
```

Reply:
Yes but for some reason it does not work.
How is original_bitwitdh computed on fresh conversion ?

Reply:
Ok so I added the assert only when I detect x.original_bit_width is <> then x.bit_width.